src/libs/elektra/keyhelpers.c

```diff
@@ -190,12 +190,17 @@ int keyNameIsSystem (const char * name)
 /**
  * @internal
  *
+ * @pre key->meta must be NULL or a valid keyset
+ *
  * clears key (all data members are set to zero)
+ * Initializes an empty metadata keyset if null or clears it.
  */
 int keyInit (Key * key)
```

Comment:
Do we need this keyInit at all? Can we make it static?

Reply:
It is used both in `key.c` and `keyhelpers.c`. Not sure if it's possible then, other than defining it statically in the header.

This occurence in `ksLookupByName` could be replaced by `keyNew` and `keyDel`.

https://github.com/ElektraInitiative/libelektra/blob/170cf57faba58979a79e0c2bf03ddd9545aa21e4/src/libs/elektra/keyset.c#L2489-L2496

Reply:
But `keyInit` now also handles initializing the `key->meta`, so we'd have two places where that happens. I would prefer to leave it as is and not make it static.

Reply:
Making a function static does not mean you need to duplicate something?

Reply:
Okay, for now I think we should just leave it as is. I will look into possible improvements in #3115, since some of the code that is causing problems will be removed/changed there.

Reply:
ok